### ISSUE TITLE
New-AzureRmResourceGroupDeployment virtualMachineScaleSets = overProvisionEnabled Error

### ISSUE THREAD

**Issue by krushmike** [Azure/azure-powershell#3388]
### Cmdlet(s)
New-AzureRmResourceGroupDeployment

### PowerShell Version
5.1.14393.0

### Module Version
Azure PowerShell 3.3

### OS Version
10.0.14393.0

### Description
Attempting to build a Virtual Machine Scale Set using an ARM template and New-AzureRmResourceGroupDeployment.  Everything completes without error except the Scale Set itself.  I get this error:

New-AzureRmResourceGroupDeployment : 3:13:59 PM - Resource Microsoft.Compute/virtualMachineScaleSets 'mevmss' failed with message '{
  "error": {
    "code": "InvalidParameter",
    "target": "overProvisionEnabled",
    "message": "The value of parameter overProvisionEnabled is invalid."
  }
}'

The only reference I can find to overProvisionEnabled is in the azure-sdk-for-java.  Nowhere is this set in my template.  I am following this example:

https://github.com/Youngr6/vmss-arm-template/tree/master/SampleVMSS/SampleVMSS

### Debug Output
{
      "id": "/subscriptions/removed/resourceGroups/MeVMSS/providers/Microsoft.Resources/deployments/AzureTemplate-0113-2028/operations/BC4F0D35FD2E277A",
      "operationId": "BC4F0D35FD2E277A",
      "properties": {
        "provisioningOperation": "Create",
        "provisioningState": "Failed",
        "timestamp": "2017-01-13T20:29:29.2334522Z",
        "duration": "PT1.8700913S",
        "trackingId": "7c26fdbb-16fa-4f0c-a121-696703ddb1f9",
        "serviceRequestId": "a4bae892-6a6f-463a-9f26-e328c2a403f8",
        "statusCode": "BadRequest",
        "statusMessage": {
          "error": {
            "code": "InvalidParameter",
            "target": "overProvisionEnabled",
            "message": "The value of parameter overProvisionEnabled is invalid."
          }
        },
        "targetResource": {
          "id": "/subscriptions/removed/resourceGroups/MeVMSS/providers/Microsoft.Compute/virtualMachineScaleSets/krushvmss",
          "resourceType": "Microsoft.Compute/virtualMachineScaleSets",
          "resourceName": "mevmss"
        }
      }
    },

### Script/Steps for Reproduction
Run the example posted above...

**Comment by cormacpayne** [comment]
@ravbhatnagar Hey Gaurav, would you mind taking a look at this issue?

**Comment by krushmike** [comment]
I ended up using the MS Azure provided sample as a new baseline to start.  Made some modifications to include the required properties to use a pre-customized VMDK and it worked!  Did not have the opportunity to compare the MS provided template against the template I was using, but there were obviously some differences to cause this issue.  Here is what I used:

https://github.com/Azure/azure-quickstart-templates/tree/master/201-vmss-windows-autoscale

Guess this is no longer a concern...
